akka-http-core/src/test/scala/akka/http/scaladsl/TestServer.scala

```diff
@@ -36,16 +39,16 @@ object TestServer extends App {
   implicit val fm = ActorMaterializer(settings)
   try {
     val binding = Http().bindAndHandleSync({
-      case req @ HttpRequest(GET, Uri.Path("/"), _, _, _) if req.header[UpgradeToWebSocket].isDefined =>
-        req.header[UpgradeToWebSocket] match {
+      case req @ HttpRequest(GET, Uri.Path("/"), _, _, _) if req.attribute(webSocketUpgrade).isDefined =>
```

Comment:
Does that make sense? First filter by `isDefined` and then match again? (unrelated to this particular change)